docs/gen-ai/gen-ai-metrics.md

```diff
@@ -0,0 +1,157 @@
+
+The conventions described in this section are specific to Generative AI
+applications.
+
+**Disclaimer:** These are initial Generative AI client metric instruments
+and attributes but more may be added in the future.
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Generative AI Operations](#generative-ai-operations)
+  - [Metric: `gen_ai.token.usage`](#metric-gen_aitokenusage)
+  - [Metric: `gen_ai.operation.duration`](#metric-gen_aioperationduration)
+
+<!-- tocstop -->
+
+## Generative AI Operations
+
+The following metric instruments describe Generative AI operations. An
+operation may be a request to an LLM, a function call, or some other
+distinct action within a larger Generative AI workflow.
+
+### Metric: `gen_ai.token.usage`
+
+This metric is [required][MetricRequired] when an operation involves the usage
+of tokens.
+
+<!-- semconv metric.gen_ai.token.usage(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `gen_ai.token.usage` | Histogram | `{token}` | Measures number of input and output tokens used | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.gen_ai.token.usage(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| [`gen_ai.operation.name`](/docs/attributes-registry/gen-ai.md) | string | The name of the operation being performed. | `chat`; `completion` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.system`](/docs/attributes-registry/gen-ai.md) | string | The name of the LLM foundation model vendor. | `openai` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.token.type`](/docs/attributes-registry/gen-ai.md) | string | The type of token being counted. | `input`; `output` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`server.address`](/docs/attributes-registry/server.md) | string | Server domain name if available without reverse DNS lookup; otherwise, IP address or Unix domain socket name. [1] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | `Required` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`server.port`](/docs/attributes-registry/server.md) | int | Server port number. [2] | `80`; `8080`; `443` | `Required` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`error.type`](/docs/attributes-registry/error.md) | string | Describes a class of error the operation ended with. [3] | `timeout`; `java.net.UnknownHostException`; `server_certificate_invalid`; `500` | `Conditionally Required` if the operation ended in an error | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
```

Comment:
I'm thinking about `error.type` for usage:
-  No streaming case: if there is an error in HTTP request of when parsing the response, then there is no usage information. 
- If there is streaming, we can start getting tokens and then response may fail. 

I think we need to call it out:
1. the error note one the usage metric should mention that it shows up only for streaming requests
2. that usage is not reported if error happens during non-streaming calls

Reply:
This could be a good discussion for the next sync. Beyond streaming/non-streaming, what about locally hosted models? Is it possible some vendor operations can end in error but still include a partial response?